File: frontend/src/types/comment.ts
import { User } from './auth';

export interface Comment {
    id: number;
    ticket_id: number;
    author_id: number;
    content: string;
    is_internal: boolean;
    created_at: string;
    author: User;
}

export interface CommentCreate {
    content: string;
    is_internal?: boolean;
}

export interface TicketHistory {
    id: number;
    ticket_id: number;
    actor_id: number;
    action: string;
    old_value: string | null;
    new_value: string | null;
    timestamp: string;
    actor: User;
}

File: backend/app/schemas/comment.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict
from .user import UserResponse

class CommentCreate(BaseModel):
    content: str
    is_internal: bool = False

class CommentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    ticket_id: int
    author_id: int
    content: str
    is_internal: bool
    created_at: datetime
    author: UserResponse
